release-notes/4.0.0.md

```diff
@@ -0,0 +1,55 @@
+# Cucumber-ruby 4.0.0 release notes
+
+This is the first stable Cucumber release of the 4.x series. It's the first stable release for 2 years and there are pretty big changes since 3.1.2. Let's have a look at the biggest changes introduced by this release.
```

Comment:
I think we can make this paragraph more helpful for users.

They want a one 1-2 sentence summary to get an idea of what they can expect to work and what they can expect to break if they upgrade. Omtting this summary might scare people off from upgrading, especially when we say there are "big changes".

Reply:
Would it be fair to say "upgrading from 3.x should not require any changes, but there are some deprecations (see below). We also have some exciting new features."